### PR TITLE
Fix end file

### PR DESCRIPTION
**Pull request by TheoMF**

### CODE REVIEW

File: gepetuto/generate.py [hunk 1]
```diff
@@ -20,8 +20,19 @@ def generate(tp_id: List[int], **kwargs):
 def generate_from_id(tp_id: int):
     """Find the corresponding ipynb and folder for a given tp_id."""
     folder = Path() / f"tp{tp_id}"
-    ipynb = next(Path().glob(f"{tp_id}-*.ipynb"))
-    generate_ipynb(ipynb, folder)
+    ipynbs = Path().glob(f"{tp_id}-*.ipynb")
+    for ipynb in ipynbs:
```
Comment (nim65s) [inline gepetuto/generate.py:24]:
```suggestion
    for ipynb in Path().glob(f"{tp_id}-*.ipynb"):
```

File: gepetuto/generate.py [hunk 2]
```diff
@@ -20,8 +20,19 @@ def generate(tp_id: List[int], **kwargs):
...
+
+
+def fix_end_of_content(content):
+    """Remove extra lines."""
+    while 1:
+        if content[-1] == "\n":
+            content.pop()
+        else:
+            return content
+    return None
```
Comment (nim65s) [inline gepetuto/generate.py:35]: This is basically a `content.strip()`, isn't it ? Or maybe a `content.strip('\n')` if you want to keep other white spaces than line breaks

File: gepetuto/generate.py [hunk 3]
```diff
@@ -87,4 +99,5 @@ def generate_ipynb(ipynb, folder, force_load=False):  # noqa: C901
                 elif dest is not None:
                     content.append(line)
     with ipynb.open("w") as f:
-        f.write(json.dumps(data, indent=1))
+        json.dump(data, f)
```
Comment (nim65s) [inline gepetuto/generate.py:102]:
```suggestion
        json.dump(data, f, indent=1)
```
It really helps with reviews :)

**Review by nim65s — CHANGES_REQUESTED** [review]

File: gepetuto/generate.py (hunk 2 above)
Comment (TheoMF) [reply gepetuto/generate.py:35]: I'm not sure it's possible to use strip here, content is a list of string

Comment (nim65s) [reply gepetuto/generate.py:34]: Ok, this is more clear, thanks

File: gepetuto/generate.py [hunk 4]
```diff
@@ -57,6 +67,7 @@ def generate_ipynb(ipynb, folder, force_load=False):  # noqa: C901
                             f"at line {line_number + 1}"
                         )
                         raise SyntaxError(msg)
+                    content = fix_end_of_content(content)
                     with dest.open("w") as f_out:
                         f_out.write("".join(content))
```
Comment (nim65s) [inline gepetuto/generate.py:72]: Maybe the `.strip()` can be applied here instead ?

**Review by nim65s — APPROVED** [review]
Thanks !

**Pull request merged**